gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/TheRestGroupRequestHandler.java

```diff
@@ -212,16 +212,16 @@ private ExploreRequest createRequest(
     return filterBuilder;
   }
 
-  private List<String> getExcludedValues(
+  private Set<String> getExcludedValues(
       String columnName, ExploreResponse.Builder originalResponse) {
     // GroupBy only supports columns expressions for now.
     return originalResponse.getRowBuilderList().stream()
         .map(rowBuilder -> rowBuilder.getColumnsMap().get(columnName))
         .map(Value::getString)
-        .collect(Collectors.toUnmodifiableList());
+        .collect(ImmutableSet.toImmutableSet());
```

Comment:
this was a minor bug that came up in the tests - we were allowing dupes in the exclusion filter. Using a guava immutable set (which maintains order) dedupes but still keeps order predictable.